inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/cdc/mysql/debezium/reader/BinlogSplitReader.java

```diff
@@ -204,6 +205,12 @@ private boolean shouldEmit(SourceRecord sourceRecord) {
                                 splitKeyType,
                                 sourceRecord,
                                 statefulTaskContext.getSchemaNameAdjuster());
+                // currently, we only support using binary search algorithm for a single split key.
+                if (key.length == 1) {
+                    FinishedSnapshotSplitInfo splitInfo = getSplitInfoByBinarySearch(
+                            finishedSplitsInfo.get(tableId), key);
```

Comment:
finishedSplitsInfo is always orderd? Does this https://github.com/apache/inlong/pull/8044 pr affects the order ?

Reply:
#8044  adjusts the order of assigning `SnapshotSplits`. After all the `SnapshotSplits` are read, the `FinishedSnapshotSplitInfos` will be sent to `MySqlBinlogSplit`. This PR adjusts the order of `FinishedSnapshotSplitInfos` in MySqlBinlogSplit, so the previous PR #8044 will not affect this one.